hi-ml/src/health_ml/model_trainer.py

```diff
@@ -0,0 +1,208 @@
+        Tuple[Trainer, StoringLogger]:
+    """
+    Creates a Pytorch Lightning Trainer object for the given model configuration. It creates checkpoint handlers
+    and loggers. That includes a diagnostic logger for use in unit tests, that is also returned as the second
+    return value.
+
+    :param container: The container with model and data.
+    :param resume_from_checkpoint: If provided, training resumes from this checkpoint point.
+    :param num_nodes: The number of nodes to use in distributed training.
+    :param kwargs: Any additional keyowrd arguments will be passed to the constructor of Trainer.
+    :return: A tuple [Trainer object, diagnostic logger]
+    """
+    logging.debug(f"resume_from_checkpoint: {resume_from_checkpoint}")
+    num_gpus = container.num_gpus_per_node()
+    effective_num_gpus = num_gpus * num_nodes
+    # Accelerator should be "ddp" when running large models in AzureML (when using DDP_spawn, we get out of GPU memory).
+    if effective_num_gpus > 1:
+        accelerator: Optional[str] = "ddp"
+        # Initialize the DDP plugin. The default for pl_find_unused_parameters is False. If True, the plugin prints out
+        # lengthy warnings about the performance impact of find_unused_parameters.
+        plugins = [DDPPlugin(num_nodes=num_nodes, sync_batchnorm=True,
+                             find_unused_parameters=container.pl_find_unused_parameters)]
+    else:
+        accelerator = None
+        plugins = []
+    logging.info(f"Using {num_gpus} GPUs per node with accelerator '{accelerator}'")
+    tensorboard_logger = TensorBoardLogger(save_dir=str(container.logs_folder), name="Lightning", version="")
+    loggers = [tensorboard_logger, AzureMLLogger(False)]
+    storing_logger = StoringLogger()
+    loggers.append(storing_logger)
+    # Use 32bit precision when running on CPU. Otherwise, make it depend on use_mixed_precision flag.
+    precision = 32 if num_gpus == 0 else 16 if container.use_mixed_precision else 32
+    # The next two flags control the settings in torch.backends.cudnn.deterministic and torch.backends.cudnn.benchmark
+    # https://pytorch.org/docs/stable/notes/randomness.html
+    # For the classification models, we observed only a small performance deterioration (increase in 10sec on total
+    # training time of 22min) when switching to deterministic.
+    if container.pl_deterministic:
+        deterministic = True
+        benchmark = False
+    else:
+        deterministic = False
+        benchmark = True
+
+    callbacks = []
+    if container.monitor_loading:
+        callbacks.append(BatchTimeCallback())
+    if num_gpus > 0 and container.monitor_gpu:
+        logging.info("Adding monitoring for GPU utilization")
+        callbacks.append(GPUStatsMonitor(intra_step_time=True, inter_step_time=True))  # type: ignore
+    # Add the additional callbacks that were specified in get_trainer_arguments for LightningContainers
+    if "callbacks" in kwargs:
```

Comment:
for context to my other comment - these are the optional callbacks we're interested in seeing both in training and test